stl/inc/execution

```diff
@@ -941,6 +941,27 @@ struct _Static_partition_range<_FwdIt, _Diff, false> {
     }
 };
 
+template <class _Cancel_or_completion, class _Diff>
+_NODISCARD _Static_partition_key<_Diff> _Get_next_if_not_canceled(
+    _Cancel_or_completion& _Status, _Static_partition_team<_Diff>& _Team) {
+    // Check cancellation/completion status before getting a chunk of work to process, instead of after. If we check
+    // after and a result was found in a later block than the one we are assigned, we won't notice an earlier/better
+    // result in our block.
+
+    bool _Done;
+    if constexpr (is_same_v<_Cancel_or_completion, _Cancellation_token>) {
+        _Done = _Status._Is_canceled();

```

Comment:
Combining the control flow and value domains makes my optimizer hate Spidey sense tingle. Did you double check that this didn't make codegen substantially worse? I know this is once per partition rather than once per N so it's not the most most important but still...

Reply:
> Combining the control flow and value domains makes my optimizer hate Spidey sense tingle.

For the record, I find your brain VTune far more irritating than Stephan's brain compiler ;) The codegen is clearly inferior: `_Get_next_if_not_done` isn't inlining, which probably doesn't affect perf much - like you said, its per chunk - but inflates code size substantially. I'm reverting this part of the change back to the original bugfix.